hardware/OpenCustomizationSelector/src/com/sonymobile/customizationselector/CustomizationSelectorService.java

```diff
@@ -46,16 +46,7 @@ public static void evaluateCarrierBundle(Context context) {
                     configurator.saveConfigurationKey();
                     CSLog.d(TAG, "isNewConfigurationNeeded - No new configuration.");
 
-                    if (configurator.isFirstApply()) {
-                        context.getPackageManager().setComponentEnabledSetting(new ComponentName(context, CustomizationSelectorActivity.class),
-                                PackageManager.COMPONENT_ENABLED_STATE_ENABLED, PackageManager.COMPONENT_ENABLED_STATE_ENABLED);
-
-                        if (isUserSetupComplete(context)) {
-                            context.startActivity(new Intent(Intent.ACTION_MAIN, null).addCategory(Intent.CATEGORY_HOME).addFlags(270565376));
-                        }
-                    } else {
-                        configurator.reApplyModem();
-                    }
+                    ModemSwitcher.reApplyModem(context);
```

Comment:
Shouldn't that be:
```suggestion
                    ModemSwitcher.reApplyModem(configurator.getTargetContext());
```

See my crash

Reply:
With this change it seems to work:

```
D CS-ServiceProvidersParser: Number of service providers found: 609
D CS-SimConfigId: getMappingMatch - mcc: 262 mnc: 07 for: S77.1
D CS-SimConfigId: Saving id: S77.1 - nbr matches: 1
D CS-SimConfigId: ***********************************
D CS-SimConfigId: Best SIM configuration id= S77.1
D CS-SimConfigId: ***********************************
D CS-DynamicConfigParser: Configurations: {S8765.1=192, S523.14=341, S6.10=461, S261.1=184, S523.15=341, S9105.1=611, S284.2=318, S523.12=341, S284.1=318, S523.13=341, S9087.1=170, S523.10=341, S523.11=341, S280.1=94, S137.1=513, S344.1=54, S9064.1=79, S367.1=73, S114.1=38, S57.1=3, S8663.1=408, S254.1=134, S212.1=198, S523.18=341, S523.19=341, S277.1=93, S523.16=341, S523.17=341, S273.1=903, S187.7=477, S187.5=477, S122.1=274, S187.1=477, S8655.4=483, S175.1=901, S8655.1=483, S46.1=523, S8655.7=483, S285.3=928, S285.2=928, S8655.5=483, S23.1=911, S8655.6=483, S366.6=640, S285.1=928, S366.7=640, S8683.1=88, S366.4=640, S366.5=640, S366.2=640, S176.1=906, S8626.2=350, S75.2=42, S8626.4=350, S75.1=13, S278.1=92, S33.1=286, S9116.1=492, S8773.1=311, S523.65=358, S523.66=358, S523.63=358, S523.64=358, S523.61=358, S6.9=461, S366.19=640, S523.62=358, S6.7=461, S523.60=358, S6.8=461, S6.5=461, S6.6=461, S123.1=273, S264.18=408, S687.1=442, S22.4=549, S22.3=549, S206.1=910, S687.2=441, S26.1=915, S264.10=408, S264.11=408, S264.12=408, S264.13=408, S264.14=408, S45.1=58, S264.15=408, S9112.1=452, S221.4=404, S264.16=408, S22.2=549, S221.5=404, S264.17=408, S22.1=549, S523.58=358, S9126.2=489, S523.59=358, S9126.1=489, S523.56=358, S523.57=358, S523.54=358, S523.55=358, S9089.1=193, S523.52=358, S523.53=358, S523.50=358, S523.51=358, S346.1=55, S369.1=62, S218.1=492, S59.1=611, S59.2=611, S55.3=34, S74.1=415, S256.1=420, S32.1=177, S523.47=358, S523.48=358, S523.1=341, S523.45=358, S523.2=341, S523.46=358, S252.1=133, S523.7=341, S523.8=341, S523.9=341, S523.3=341, S523.4=341, S523.5=341, S523.6=341, S173.1=306, S264.9=408, S264.5=408, S264.6=408, S264.7=408, S264.8=408, S44.1=59, S21.1=122, S6552.1=191, S249.1=196, S264.1=408, S9111.1=446, S264.2=408, S523.49=358, S264.3=408, S264.4=408, S260.1=41, S283.1=426, S8760.2=929, S136.1=37, S9065.2=114, S537.1=244, S120.1=71, S77.3=310, S58.1=326, S219.1=424, S253.1=132, S8760.1=929, S77.2=310, S31.1=39, S77.1=310, S54.1=2, S257.1=272, S12.1=297, S8960.1=483, S545.2=308, S8779.1=567, S545.1=308, S8960.2=483, S523.21=341, S523.22=341, S523.20=341, S125.1=40, S8779.6=567, S174.1=909, S284.4=318, S20.1=311, S284.3=318, S43.1=57, S284.5=318}
D CS-ModemConfParser: setupFilePaths - configId: 310
D CS-ModemConfParser: setupFilePaths - path: /system/etc/customization/modem/310/modem.conf
D CS-ModemConfParser: Parsed modem: 'telefonica_germany_volte_vilte_vowifi'
I CS-CommonUtil: Returning bundle with sim id S77.1, modem: telefonica_germany_volte_vilte_vowifi, config id: 310
D CS-Configurator: Direct Boot is enabled. Use device encrypted storage.
D CS-Configurator: OldConfigKey: 1310-6690R4C1307-751147.2.A.11.22818949227211205726581
D CS-CommonUtil: getDefaultSubId: 2
D CS-Configurator: getIccid: 8949227211205726581
D CS-Configurator: CurrentConfKey: 1310-6690R4C1307-751147.2.A.11.22818949227211205726581
D CS-Configurator: isNewConfigurationNeeded - ConfigKey not updated, no need to evaluate
D CS-CommonUtil: getDefaultSubId: 2
D CS-Configurator: getIccid: 8949227211205726581
D CS-Configurator: CurrentConfKey: 1310-6690R4C1307-751147.2.A.11.22818949227211205726581
D CS-Configurator: Direct Boot is enabled. Use device encrypted storage.
D CS-Configurator: saveConfigKey - key saved: 1310-6690R4C1307-751147.2.A.11.22818949227211205726581
D CS-CustomizationSelectorService: isNewConfigurationNeeded - No new configuration.
D CS-Configurator: Direct Boot is enabled. Use device encrypted storage.
D CS-ModemSwitcher: Read line: 0,amss_fsg_lilac_tar.mbn
D CS-ModemSwitcher: Target filename of: amss_fsg_lilac_tar.mbn is: amss_fsg_lilac_tar.mbn
D CS-ModemSwitcher: reApplyModem - current modem: /system/etc/customization/modem/amss_fsg_lilac_tar.mbn
D CS-ModemSwitcher: reApplyModem - Re-writing 2405 with modem amss_fsg_lilac_tar.mbn
I CS-ModemSwitcher: reApplyModem - 2405 was re-written successfully
I CS-ModemSwitcher: reApplyModem - Modem Switcher 2404 cleared
```

Reply:
That's great ! :D